test/WebJobs.Extensions.Tests/Extensions/Http/ClaimsPrincipalEndToEndTests.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.Tests.Common;
+using Microsoft.Azure.WebJobs.Host.Timers;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Tests.Extensions.Http
+{
+    [Trait("Category", "E2E")]
+    public class ClaimsPrincipalEndToEndTests
+    {
+        private const string UserAuthType = "aad";
+        private const string UserClaimType = "name";
+        private const string UserClaimValue = "Sample";
+        private const string Name = "sample@microsoft.com";
+        private const string UserIdentityJson = @"{
+  ""auth_typ"": """ + UserAuthType + @""",
+  ""claims"": [
+    {
+      ""typ"": """ + UserClaimType + @""",
+      ""val"": """ + UserClaimValue + @"""
+    },
+    {
+      ""typ"": ""http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"",
+      ""val"": """ + Name + @"""
+    },
+    {
+      ""typ"": ""http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn"",
+      ""val"": """ + Name + @"""
+    }
+  ],
+  ""name_typ"": ""http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"",
+  ""role_typ"": ""http://schemas.microsoft.com/ws/2008/06/identity/claims/role""
+}";
+
+        private const string KeyAuthType = "key";
+        private const string KeyIdClaimType = "urn:functions:keyId";
+        private const string KeyIdClaimValue = "master";
+        private const string KeyLevelClaimType = "urn:functions:authLevel";
+        private const string KeyLevelClaimValue = "admin";
+        private const string KeyIdentityJson = @"{
+  ""auth_typ"": """ + KeyAuthType + @""",
+  ""claims"": [
+    {
+      ""typ"": """ + KeyIdClaimType + @""",
+      ""val"": """ + KeyIdClaimValue + @"""
+    },
+    {
+      ""typ"": """ + KeyLevelClaimType + @""",
+      ""val"": """ + KeyLevelClaimValue + @"""
+    }
+  ],
+  ""name_typ"": ""http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"",
+  ""role_typ"": ""http://schemas.microsoft.com/ws/2008/06/identity/claims/role""
+}";
+
+        private const string OutputTestPath = @"webjobs_extensionstests\filebindinge2e_output";
+
+        private static ClaimsPrincipal globalPrincipal;
```

Comment:
>globalPrincipal [](start = 39, length = 15)

Is it guaranteed that these tests only run one-at-a-time?

Reply:
I'm not positive, but after talking with Mike it sounds like the E2E tests run sequentially to avoid some bugs that come up. 

This is the method I've seen used in our "EndtoEnd" in BYOB extensions (mainly the method I used in Graph tests and seen used in the EventGrid extension). It allows you to keep more of the actual test logic in the test itself, since you have the actual bound object to test against. Assuming the tests are run in parallel, I would prefer to keep the tests this way.

---
In reply to: [181945285](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/404#discussion_r181945285) [](ancestors = 181945285)